contrib/objcwrapper/src/objc_generator.nit

```diff
@@ -20,29 +20,69 @@ import opts
 import objc_model
 
 redef class Sys
+
 	# Path to the output file
 	var opt_output = new OptionString("Output file", "-o")
-end
 
-class CodeGenerator
-	# Merge the calls to `alloc` and `init...` in a single constructor?
+	# Shall `init` methods/constructors be wrapped as methods?
 	#
-	# If `true`, also the default behavior, initializing an extern Objective-C object looks like:
+	# By default, these methods/constructors are wrapped as extern constructors.
+	# So initializing an extern Objective-C object looks like:
 	# ~~~nitish
 	# var o = new NSArray.init_with_array(some_other_array)
 	# ~~~
 	#
-	# If `false`, the object must first be allocated and then initialized.
+	# If this option is set, the object must first be allocated and then initialized.
 	# This is closer to the Objective-C behavior:
 	# ~~~nitish
 	# var o = new NSArray
 	# o.init_with_array(some_other_array)
 	# ~~~
-	var init_with_alloc = true is writable
+	var opt_init_as_methods = new OptionBool(
+		"Wrap `init...` constructors as Nit methods instead of Nit constructors",
+		"--init-as-methods")
+
+	private var nit_to_java_types: Map[String, String] is lazy do
+		var types = new HashMap[String, String]
+		types["char"] = "Byte"
+		types["short"] = "Int"
+		types["short int"] = "Int"
+		types["int"] = "Int"
+		types["long"] = "Int"
+		types["long int"] = "Int"
+		types["long long"] = "Int"
+		types["long long int"] = "Int"
+		types["float"] = "Float"
+		types["double"] = "Float"
+		types["long double"] = "Float"
+
+		types["NSUInteger"] = "Int"
+		types["BOOL"] = "Bool"
+		types["id"] = "NSObject"
+		types["constid"] = "NSObject"
+		types["SEL"] = "NSObject"
```

Comment:
It's probably the method id...

~~~Objective-C
SEL foo_id = NSSelectorFromString(@"foo")
[receiver performSelector:foo_id]
~~~